src/NativeScript/JSErrors.mm

```diff
@@ -73,6 +73,34 @@ void reportFatalErrorBeforeShutdown(ExecState* execState, Exception* exception,
             }
             std::cerr << "\n";
         }
+        
+        //OBSERVATION:
```

Comment:
I would like to see this observation validated.
We can leave it as technical debt (although I really don't like this option) but we must know what is the reason for this behavior in JSC.
In my mind this is an issue that was there for at least an year. It can't be so critical that we would like to merge it immediately without knowing what is the specific reason for this behavior.

Reply:
@zahhak - in terms of behaviour the observation is validated. The open question here is whether this is an issue with JSC or an issue with our promise-based implementation for module resolution. I agree that we can spend some more time debugging here and decide if we want to release this as a patch or talk to Apple about it and do it afterwards.